### PR TITLE
Add date filtering to subscriptions

### PR DESCRIPTION
**Pull request by rudigiesler**
This PR will add the ability to filter on created_at date ranges for subscriptions, through query params.

### CODE REVIEW

File: subscriptions/views.py [hunk 1]
```diff
@@ -1,26 +1,40 @@
...
+class SubscriptionFilter(filters.FilterSet):
+    created_from = django_filters.IsoDateTimeFilter(
+        name="created_at", lookup_type="gte")
+    created_to = django_filters.IsoDateTimeFilter(
+        name="created_at", lookup_type="lte")
+
+    class Meta:
+        model = Subscription
+        filters = (
+            'identity', 'messageset_id', 'lang', 'active', 'completed',
```
Comment (erikh360) [inline subscriptions/views.py]: I think you need a filters list here, like this:
https://github.com/praekelt/seed-message-sender/blob/develop/message_sender/views.py#L71

File: subscriptions/views.py [hunk 2]
```diff
@@ -1,26 +1,40 @@
...
+import django_filters
 
 from .models import Subscription
 from .serializers import SubscriptionSerializer, CreateUserSerializer
 from .tasks import send_next_message, scheduled_metrics
 from seed_stage_based_messaging.utils import get_available_metrics
 
 
+class SubscriptionFilter(filters.FilterSet):
+    created_from = django_filters.IsoDateTimeFilter(
```
Comment (erikh360) [inline subscriptions/views.py]: Any reason for not making this `created_before` and `created_after`?

File: subscriptions/views.py (hunk 1 above)
Comment (rudigiesler) [inline subscriptions/views.py]: Ah, that's a typo that has unintended consequences that cause the tests to pass. It seems like if `fields` and `exclude` aren't present, then it allows you to filter on all fields, which is pretty much what we're specifying here. https://github.com/carltongibson/django-filter/blob/develop/django_filters/filterset.py#L252-L253 .

So I'm going to remove this to allow us to filter on all fields.

File: subscriptions/views.py (hunk 2 above)
Comment (rudigiesler) [inline subscriptions/views.py]: This is what the identity store uses, so I assumed that that was the standard for the seed stack, and wanted to keep it the same as everywhere else. I'm not actually sure if this is really the case.

**Comment by rudigiesler** [comment]
@erikh360 Ready for another review.

**Pull request merged**